### PR TITLE
Crypto Master - Mobile Version - React Redux Capstone

### PR DESCRIPTION
**Pull request by Elerqsousy**
1. Setup project and file structure
- Setup project using `create-react-app`
- Setup linters
- Create a descriptive readme file
- Change the title and add a customized logo

2. Build nav skeleton and search component
- Used React-Router-Dom to create routes
- Used Routes to build Nav structure
- Structured page Routes in the app component.

3. Build redux core functionalities
- Configured Redux Store using `configureStore` from Redux Toolkit
- Structure reducers and actions using slices
- Create Async actions using `createAsyncThunk`

4. Build Components
- Built component view to present data
- Performed maintenance required on data after configuration
- Added components integration logic

5. Add styling

6. Add tests and Fix persisting errors
- Added unit testing for pure functions in redux
- Added integration tests for the app using the `React Testing Library` and a duplicate mocked redux store.

7. Testing and error fixing
- fix conditional rendering issues
- Fix deployment and console errors

<hr>

[Deployment](https://adorable-creponne-58a432.netlify.app) is hosted by Netlify.
[Live Demo](https://drive.google.com/file/d/1hxOiBF3in3UUTZhqJ2M7Upx_QKrMUH-E/view?usp=sharing)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,51 @@
...
+# Crypto Master
+This project is a mobile and web compatible app for cryptocurrency enthusiasts, where they can access, view, and compare up-to-date information about all sorts of cryptocurrencies. Their is a dedicated profile page for viewing and customizing personal and security options and data, and more importantly have access to their pre assigned favourites and saves.
 
-## Available Scripts
+## Screensot
+
+![screenshot](./Screenshot1.png)
+![screenshot](./Screenshot2.png)
+![screenshot](./Screenshot3.png)
+![screenshot](./Screenshot4.png)
```
Comment (Whoistolu) [inline README.md]: - [x] [OPTIONAL] Kindly ensure you keep your project image in a particular directory in order to have a well organised project.

File: README.md [hunk 2]
```diff
@@ -1,8 +1,51 @@
-# Getting Started with Create React App
+[](https://img.shields.io/badge/Microverse-blueviolet)
 
-This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+# Crypto Master
+This project is a mobile and web compatible app for cryptocurrency enthusiasts, where they can access, view, and compare up-to-date information about all sorts of cryptocurrencies. Their is a dedicated profile page for viewing and customizing personal and security options and data, and more importantly have access to their pre assigned favourites and saves.
```
Comment (Whoistolu) [inline README.md:4]: - [x] [REQUIRED] Kindly ensure this project contains an "Acknowledgements" section that gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/) as stated in the project requirement.

File: src/components/navigation/Nav.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { Link, NavLink } from 'react-router-dom';
+import { v4 as uuid } from 'uuid';
+import logo from '../../assets/colored-logo.png';
+import routLinks from './NavRoutes';
+import SearchBar from './SearchBar';
+import styles from './navigation.module.css';
+
+export default function Nav() {
```
Comment (Whoistolu) [inline src/components/navigation/Nav.js:9]: - [x] [OPTIONAL] In order to have a better UI, let's ensure the items in the nav have a yellow color as you have it in the presentation video.
Screenshot from 2022-06-04 20-47-23

File: src/pages/details/index.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+  const {
+    description,
+    id,
+    name,
+    img,
+    links,
+    ChnagePercentage,
+    currentPrice,
+  } = singleCoin;
```
Comment (Whoistolu) [inline src/pages/details/index.js:24]: - [x] [OPTIONAL] Please update your deployed version as we have some slight differences, take for example the pages description is different from what I can see on the local version.

Local Version  |  Deployed Version
:-------------------------:|:-------------------------:

Screenshot from 2022-06-04 21-06-20 | Screenshot from 2022-06-04 21-07-12

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @Elerqsousy 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [x]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (mateo951) [reply README.md:4]: Addressed 🔥

**Review by mateo951 — APPROVED** [review]
### Hi @Elerqsousy, 

Your project is complete!  There is nothing else to say other than... it's time to merge it ✔ 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**